api/rosetta/construction.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package rosetta
+
+const (
+	// FlowSignatureAlgorithm is the signing algorithm supported by Flow.
+	FlowSignatureAlgorithm = "ecdsa"
```

Comment:
Not really, at least I did not find any.

From what I found, Flow uses ECDSA algorithm, and supports ECDSA on NIST P-256 curve and ECDSA on secp256k1 curve ([source](https://docs.onflow.org/flow-go-sdk/generating-keys/)).
There are constants for those specific cases:
https://pkg.go.dev/github.com/onflow/flow-go-sdk/crypto#pkg-constants


For the field for Rosetta though, it seems that it only expects one of `ecdsa`, `ecdsa_recovery`, `ed25519`, `schnorr_1` or `schnorr_poseidon` ([source](https://www.rosetta-api.org/docs/models/SignatureType.html)).

I'm not sure how useful this field is like this since it's not 100% precise, but I figured it might be good form to populate it. It is optional though ([source](https://www.rosetta-api.org/docs/models/SigningPayload.html))

Reply:
I guess we can leave it like that, then. Though you are right, it's not super precise. Ed25519, for instance, is also ECDSA, and I think they are confounding algorithm and curves.